buildalias/buildalias.go

```diff
@@ -0,0 +1,201 @@
+package main;
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+type GameServerBuild struct {
+	BuildId string
+	Weight int
+}
+
+func main() {
+	args := os.Args;
+	fileName := "thundernetes-buildalias.yaml";
+	helpMessage := "\n\tThundernetes build alias feature\n\n" +
+	"1) create - generates a new build alias with the name provided." +
+	" Each alias is made of one or more pairs of build id and weights (priority of assignment for allocation). Template for using:" +
+	"\n\n\tbuildalias create <alias-name> <buildId1> <weight1> <buildId2> <weight2> ... <buildIdN> <weightN>\n\n" +
+	"2) allocate - create a new game server instance for the build alias and session id provided. Template for using\n\n\t" +
+	"buildalias allocate <alias-name> <sessionId>\n\n" +
+	"3) help - Displays this message and exit";
+
+	if len(args) == 1 {
+		fmt.Println(helpMessage);
+		return;
+	}
+
+	switch(args[1]) {
+
+	case "create":
+		CreateBuildAlias(args, fileName);
+
+	case "allocate":
+		AllocateForBuildAlias(args, fileName);
+
+	case "help":
+		fmt.Println(helpMessage);
+
+	default:
+		fmt.Println("Sorry, but the command "+ args[1] + " is not recognized.");
+	}
+}
+
+func CreateBuildAlias(args []string, fileName string) {

```

Comment:
we should add unit tests for all the methods in this file

Reply:
Where can I add these unit tests? should I use google test suit?

Reply:
you can add a file with the _test postfix (e.g. main_test.go) which includes the unit tests -> https://gobyexample.com/testing